bowling-game/tabcorp/31-05-2017/test.js

```diff
@@ -0,0 +1,53 @@
+const assert = require('assert');
+
+describe('Bowling Game', () => {
+
+  const createGame = () => {
+    let rollsResult = []
+    return {
+      roll: function(pinsTakenDown) {
+        rollsResult.push(pinsTakenDown)
+      },
+      score: function() {
+        const pinsTakenDownToRolls = function(pinsTakenDown) {
+          const roll = {
+            pinsTakenDown: pinsTakenDown,
+            strike: false
+          }
+          if (pinsTakenDown === 10) {
+            roll.strike = true
+          }
+          return roll;
+        }
+        const toSumOfPinsTakenDown = (previousSum, currentRoll) => {
+          return previousSum + currentRoll.pinsTakenDown;
+        }
+
+        const rolls = rollsResult.map(pinsTakenDownToRolls)
+        return rolls.reduce(toSumOfPinsTakenDown, 0);
```

Comment:
Interesting… as the problem statement defines the interface, `roll` and `score` (which are the `mutator` and `accessor` methods respectively), it is great that we can take the opportunity to introduce new concepts/terminologies. 👍

<sub>***Small note*** (literally :-)): What we see here is the applicability of CQS. Although similar in nature, CQS and CQRS are _slightly_ different.</sub>